Skip tests when there is no API Key

sharoonthomas opened the pull request:
This can be used to run tests when PRs are sent

prakashpp commented on tests/test_fulfil_client.py:
```diff
@@ -7,10 +7,22 @@
 import pytest
 from fulfil_client import Client, ClientError, ServerError
 
+need_api_key = pytest.mark.skipif(
```
this is already defined in coftest

prakashpp reviewed (changes requested):
Build is failing for python 3

sharoonthomas committed:
Make implementation of search_read_all more efficient

Paginating through a database is more expensive for
large records sets. In this case, read all the ids
at once and then page through them locally and call
read with the ids instead.
